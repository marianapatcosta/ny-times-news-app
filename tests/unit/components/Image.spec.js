import { mount } from "@vue/test-utils";
import ImageComponent from "@/components/image/Image.vue";

describe("Image.vue", () => {
  const image = new Image();
  const src = image.src;
  const alt = "image description";
  const handleLoad = jest.fn();
  let wrapper;
  beforeEach(() => {
    wrapper = mount(ImageComponent, {
      props: {
        props: { src, alt },
        methods: { handleLoad },
      },
    });
  });

  afterEach(() => {
    wrapper.unmount();
  });
  it("should render without errors and match snapshot renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should not show image before loading", async () => {
    const image = wrapper.find("img");
    expect(image.isVisible()).toBe(false);
  });

  it("should show image after loading", async () => {
    const image = wrapper.find("img");
    await image.trigger("load");
    expect(image.isVisible()).toBe(true);
  });
});
