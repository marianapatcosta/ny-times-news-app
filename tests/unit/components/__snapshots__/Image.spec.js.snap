// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image.vue should render without errors and match snapshot renders correctly 1`] = `
<div
  class="image__wrapper"
  methods="[object Object]"
  props="[object Object]"
>
  <div
    class="image__placeholder"
  />
  <img
    class="image"
    loading="lazy"
    style="display: none;"
  />
</div>
`;
